weave-js/src/components/PagePanelComponents/Home/Browse3/datasets/schemaUtils.ts

```diff
@@ -589,21 +588,16 @@ export const mapCallsToDatasetRows = (
           mapping.sourceField
         );
       }
-
       if (sourceValue !== undefined) {
         if (
           typeof sourceValue === 'object' &&
           sourceValue !== null &&
           !Array.isArray(sourceValue)
         ) {
-          // Flatten nested objects into path-based keys
-          const flattenedValues = flattenObject(sourceValue);
-          Object.keys(flattenedValues).forEach(key => {
-            const fullKey = `${mapping.targetField}.${key}`;
-            rowData[fullKey] = flattenedValues[key];
+          Object.entries(flattenObject(sourceValue)).forEach(([key, value]) => {
+            rowData[`${mapping.targetField}.${key}`] = value;
```

Comment:
Check for prefixes in the schema, eg include `inputs.examples.row` if `inputs.example` was mapped